### PR TITLE
[WIP] Deploy latest version of BMO

### PR DESCRIPTION
**Pull request by yprokule**

### CODE REVIEW

**Comment by hardys** [comment]
I don't think this will work without the other related changes tracked via https://github.com/openshift-metal3/dev-scripts/issues/548 ?

**Comment by yprokule** [comment]
Worked for me locally, but I agree that more thorough testing is needed. Thanks @hardys !
```
oc describe baremetalhost openshift-worker-rhhi-node-3 -n openshift-machine-api
Name:         openshift-worker-rhhi-node-3
Namespace:    openshift-machine-api
Labels:       <none>
Annotations:  kubectl.kubernetes.io/last-applied-configuration:
                {"apiVersion":"metal3.io/v1alpha1","kind":"BareMetalHost","metadata":{"annotations":{},"name":"openshift-worker-rhhi-node-3","namespace":"...
API Version:  metal3.io/v1alpha1
Kind:         BareMetalHost
Metadata:
  Creation Timestamp:  2019-05-21T06:48:17Z
  Finalizers:
    baremetalhost.metal3.io
  Generation:        2
  Resource Version:  210219
  Self Link:         /apis/metal3.io/v1alpha1/namespaces/openshift-machine-api/baremetalhosts/openshift-worker-rhhi-node-3
  UID:               6a2a5f4b-7b94-11e9-b305-5254004edacc
Spec:
  Bmc:
    Address:           ipmi://192.168.123.1:6233
    Credentials Name:  openshift-worker-rhhi-node-3-bmc-secret
  Boot MAC Address:    52:54:00:6e:10:83
  Description:         
  Hardware Profile:    
  Online:              true
Status:
  Error Message:  
  Good Credentials:
    Credentials:
      Name:               openshift-worker-rhhi-node-3-bmc-secret
      Namespace:          openshift-machine-api
    Credentials Version:  209344
  Hardware:
...
  Last Updated:        2019-05-21T06:51:26Z
  Operational Status:  OK
  Powered On:          true
  Provisioning:
    ID:  38701c8d-6b16-4492-aff5-45ca2cc85ae5
    Image:
      Checksum:  
      URL:       
    State:       ready
Events:
  Type    Reason              Age   From                         Message
  ----    ------              ----  ----                         -------
  Normal  Registered          166m  metal3-baremetal-controller  Registered new host
  Normal  BMCAccessValidated  166m  metal3-baremetal-controller  Verified access to BMC
  Normal  InspectionStarted   166m  metal3-baremetal-controller  Hardware inspection started
  Normal  InspectionComplete  164m  metal3-baremetal-controller  Hardware inspection completed
  Normal  ProfileSet          164m  metal3-baremetal-controller  Hardware profile set: unknown
  Normal  PowerOn             163m  metal3-baremetal-controller  Host powered on
```
```
 oc describe po metal3-baremetal-operator-599c685865-p2pzk -n openshift-machine-api
Name:               metal3-baremetal-operator-599c685865-p2pzk
Namespace:          openshift-machine-api
Priority:           0
PriorityClassName:  <none>
Node:               rhhi-node-0/192.168.123.115
Start Time:         Tue, 21 May 2019 06:35:24 +0000
Labels:             name=metal3-baremetal-operator
                    pod-template-hash=599c685865
Annotations:        k8s.v1.cni.cncf.io/networks-status:
                      [{
                          "name": "openshift-sdn",
                          "interface": "eth0",
                          "ips": [
                              "10.129.0.48"
                          ],
                          "default": true,
                          "dns": {}
                      }]
Status:             Running
IP:                 10.129.0.48
Controlled By:      ReplicaSet/metal3-baremetal-operator-599c685865
Containers:
  baremetal-operator:
    Container ID:  cri-o://4f412ee3e0a8bb542704fe3d579da08e86bc2369880420ae18c4ea7c2dc8a607
    Image:         quay.io/metal3-io/baremetal-operator:master
    Image ID:      quay.io/metal3-io/baremetal-operator@sha256:31a741b93888216d67ea4349a7b6b2ef6057fcbb93c9befafe2518433fe53087
    Port:          60000/TCP
    Host Port:     0/TCP
    Command:
      /baremetal-operator
    State:          Running
      Started:      Tue, 21 May 2019 06:35:38 +0000
    Ready:          True
    Restart Count:  0
    Environment:
      WATCH_NAMESPACE:            openshift-machine-api (v1:metadata.namespace)
      POD_NAME:                   metal3-baremetal-operator-599c685865-p2pzk (v1:metadata.name)
      OPERATOR_NAME:              baremetal-operator
      DEPLOY_KERNEL_URL:          http://172.22.0.1/images/ironic-python-agent.kernel
      DEPLOY_RAMDISK_URL:         http://172.22.0.1/images/ironic-python-agent.initramfs
      IRONIC_ENDPOINT:            http://localhost:6385/v1/
      IRONIC_INSPECTOR_ENDPOINT:  http://localhost:5050/v1/
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from metal3-baremetal-operator-token-g222f (ro)
  ironic-proxy:
    Container ID:  cri-o://9c780981862cd78df36b6c75891d4659a0f14cc42340d7fb6d457758657893da
    Image:         alpine/socat
    Image ID:      docker.io/alpine/socat@sha256:5f245d7a2d63fccdb098834d00d9fb04c404a3f1423eb2f84045fc00e93d7c32
    Port:          <none>
    Host Port:     <none>
    Command:
      socat
      tcp-listen:6385,fork,reuseaddr
      tcp-connect:172.22.0.1:6385
    State:          Running
      Started:      Tue, 21 May 2019 06:35:42 +0000
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from metal3-baremetal-operator-token-g222f (ro)
  ironic-inspector-proxy:
    Container ID:  cri-o://d4707caf0cc778644bea5905732cbfeae2fc98bf8cc2bc3d5b3d6b93b16cca8a
    Image:         alpine/socat
    Image ID:      docker.io/alpine/socat@sha256:5f245d7a2d63fccdb098834d00d9fb04c404a3f1423eb2f84045fc00e93d7c32
    Port:          <none>
    Host Port:     <none>
    Command:
      socat
      tcp-listen:5050,fork,reuseaddr
      tcp-connect:172.22.0.1:5050
    State:          Running
      Started:      Tue, 21 May 2019 06:35:46 +0000
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from metal3-baremetal-operator-token-g222f (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  metal3-baremetal-operator-token-g222f:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  metal3-baremetal-operator-token-g222f
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:          <none>
```

File: 08_deploy_bmo.sh [hunk 1]
```diff
@@ -7,21 +7,17 @@ source logging.sh
...
 # Get the latest bits for baremetal-operator
-export BMOPATH="$GOPATH/src/github.com/metalkube/baremetal-operator"
+export BMOPATH="$GOPATH/src/github.com/metal3-io/baremetal-operator"
 
 # Make a local copy of the baremetal-operator code to make changes
 cp -r $BMOPATH/deploy ocp/.
 sed -i 's/namespace: .*/namespace: openshift-machine-api/g' ocp/deploy/role_binding.yaml
-# FIXME(dhellmann): Use the pre-rename operator until this repo
-# works with the renamed version.
-# - https://github.com/openshift-metal3/dev-scripts/issues/548
```
Comment (russellb) [inline 08_deploy_bmo.sh]: This issue is tracking when we can move to the latest BMO.

**Comment by yprokule** [comment]
Closed via 5e01753833c03c5dc2c3cb398853936049348b63

**Pull request closed without merging**